code/modules/food_and_drinks/food/snacks.dm

```diff
@@ -119,46 +119,46 @@
 	if((slices_num <= 0 || !slices_num) || !slice_path)
 		return 0
 
-	var/inaccurate = 0
-	if( \
-			istype(W, /obj/item/weapon/kitchen/knife) || \
-			istype(W, /obj/item/weapon/scalpel) \
-		)
-	else if( \
-			istype(W, /obj/item/weapon/circular_saw) || \
-			istype(W, /obj/item/weapon/melee/energy/sword/saber) && W:active || \
-			istype(W, /obj/item/weapon/melee/energy/blade) || \
-			istype(W, /obj/item/weapon/shovel) || \
-			istype(W, /obj/item/weapon/hatchet) \
-		)
-		inaccurate = 1
-	else if(W.w_class <= WEIGHT_CLASS_SMALL && istype(src,/obj/item/weapon/reagent_containers/food/snacks/sliceable))
+	if(W.sharp) // First check if the object is sharp, we're probably trying to cut the food.
```

Comment:
I'm going to recommend you add a prompt to ask if the player wants to slice or stash the object here, so you can hide a shiv in a cake or use it to cut the cake at your discretion.

Reply:
Yeah I originally thought about going along that route, but I'm not a fan of too many prompts, especially on this kind of action where the chef could potentially line up a bunch of things to slice and then have to click a prompt every time. If possible it would be nice if there was a more intuitive way to do it. Maybe with a verb linked to a shortcut such as alt click for putting objects inside? What do you think?

Reply:
Honestly, the more i look at this, the more i wonder why the stashing of items isn't specifically on the `/obj/item/weapon/reagent_containers/food/snacks/sliceable` path, as that bit specifically looks for that type to even offer the option.

Reply:
I imagine `/obj/item/weapon/reagent_containers/food/snacks/sliceable` was probably just added later to the pre-existing code and slicing and stashing was just bulked in the one big `attackby()`.